### ISSUE TITLE
Promote --experimental_cc_implementation_deps to not be experimental

### ISSUE THREAD

**Issue by limdor** [bazelbuild/bazel#20098]
### Description of the feature request:

This feature is flipped by default in master and also in 6.4. It would make sense to remove the experimental name from it.

During BazelCon 2023 I had a 1 to 1 with people from the C++ team and I understood that they are not aware of any current blocker in order to remove the experimental part of the name.

### Which category does this issue belong to?

C++/Objective-C Rules

**Comment by martis42** [comment]
Any update on this?
If there is right now no free developer capacity to perform the actual flag rename, that is fine.
It would however be great to have an official statement if this flag will become non experimental or not and if we can start relying on it. Especially, given this experimental feature is now active by default for all users of recent Bazel versions.

**Comment by iancha1992** [comment]
cc: @comius

**Comment by limdor** [comment]
Any news on this? Can someone from the Bazel team give an statement? Thanks

**Comment by fmeum** [comment]
This is in Bazel 7: https://github.com/bazelbuild/bazel/commit/05787f3f8bc3f3866ccb4d7650b331373a053640

**Comment by limdor** [comment]
This is not correct, the flip is done but experimental is still on the name. Please reopen the ticket @fmeum

**Comment by limdor** [comment]
You can also see that in the documentation is still talking about the experimental flag https://bazel.build/reference/be/c-cpp#cc_library.implementation_deps and this gives the impression it is experimental regardless of being flip or not.

**Comment by fmeum** [comment]
I reopened the issue and will look into updating the docs. A flag being labeled experimental doesn't necessarily mean that the functionality itself is still considered experimental though, only that toggling the flag is.

**Comment by fmeum** [comment]
@pzembrod Is `--experimental_cc_implementation_deps` enabled at Google?

**Comment by martis42** [comment]
> A flag being labeled experimental doesn't necessarily mean that the functionality itself is still considered experimental though, only that toggling the flag is.

From the perspective of users, this is a confusing definition of _exprimental_.

**Comment by limdor** [comment]
If what @fmeum is correct it would be good to be documented somewhere. From the metadata tags explanation it seems different.

> This option triggers an experimental feature with no guarantees of functionality.

**Comment by martis42** [comment]
@fmeum I gave removing the flag a shot and the CI seems to be happy: https://github.com/bazelbuild/bazel/pull/23010

**Comment by martis42** [comment]
In my opinion this is still a consistency issue in Bazel's CLI

**Comment by jmillikin** [comment]
Gentle ping, the `implementation_deps` attribute is still marked experimental and isn't available to use via `cc_common.compile`.

https://github.com/bazelbuild/rules_cc/blob/541eda5d72e9b5f18e6a24e8d14975afcd865717/cc/private/cc_common.bzl#L499-L505

```starlark
    if module_map != _UNBOUND or \
       # ...
       implementation_compilation_contexts != _UNBOUND or \
       # ...
       non_compilation_additional_inputs != _UNBOUND:
        _cc_internal.check_private_api(allowlist = _PRIVATE_STARLARKIFICATION_ALLOWLIST)
```

I assume the `@rules_cc` code can be updated once the Bazel side marks that attribute as no longer experimental?
